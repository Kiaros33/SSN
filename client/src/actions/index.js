import axios from 'axios';


/*------------------------------------------ USER ACTIONS ------------------------------------------*/


// Login //
export function login({email,password}){

    const request = axios.post('/api/login',{email,password})
    .then(response=>response.data);

    return {
        type:'USER_LOGIN',
        payload:request
    }
}


// Register with JWT //
export function register(user) {
    const request = axios.post(`/api/register`,user)
    .then(response=>response.data);

    return {
        type:'USER_REGISTER',
        payload:request
    }
}


// Check is Auth //
export function isAuth(){

    const request = axios.get('/api/isAuth')
    .then(response=>response.data);

    return {
        type:'USER_AUTH',
        payload:request
    }
}


// Edit user //
export function editUser(user){
    let file = user.file;
    let fileName = user.fileName;
    const request = axios.post(`/api/editUser`,user)

    return (dispatch)=>{
        request.then(({data})=>{
            let user = data;
            const fd = new FormData();
            fd.append('image',file,fileName);
            let config = {
                headers: {'Content-Type':'multipart/form-data'}
            }
            axios.post(`/api/upload`,fd,config)
            .then(({data})=>{
                let response = {
                    success:true,
                    isAuth:true,
                    id:user.id,
                    nickname:user.nickname,
                    email:user.email,
                    image:user.image
                }

                dispatch({
                    type:'USER_EDIT',
                    payload:response
                })
            })
        })
    }
}


// Search for friend by e-mail //
export function friendSearch(email){
    const request = axios.get(`/api/friendSearch?email=${email}`)
    .then(response=>response.data);
    return {
        type:'FRIEND_SEARCH',
        payload:request
    }
}


// Invite friend //
export function friendInvite(userOut,userIn){
    const request = axios.post(`/api/friendInvite`,{userOut,userIn})
    .then(response=>response.data);
    return {
        type:'FRIEND_INVITE',
        payload:request
    }
}


// Get all outgoing requests //
export function getAllOut(userId){
    const request = axios.get(`/api/getAllOut?userId=${userId}`)
    .then(response=>response.data);
    return {
        type:'GET_ALL_OUT',
        payload:request
    }
}


// Get all incoming requests //
export function getAllIn(userId){
    const request = axios.get(`/api/getAllIn?userId=${userId}`)
    .then(response=>response.data);
    return {
        type:'GET_ALL_IN',
        payload:request
    }
}


// Cancel outgoing request //
export function cancelOut(curUser,reqUser){
    const request = axios.post(`/api/cancelOut`,{curUser,reqUser})
    .then(response=>response.data);
    return {
        type:'CANCEL_OUT_REQ',
        payload:request
    }
}


// Reject incoming request //
export function cancelIn(curUser,reqUser){
    const request = axios.post(`/api/cancelIn`,{curUser,reqUser})
    .then(response=>response.data);
    return {
        type:'CANCEL_IN_REQ',
        payload:request
    }
}


// Accept incoming request (become friends) //
export function acceptIn(curUser,reqUser){
    const request = axios.post(`/api/acceptIn`,{curUser,reqUser})
    .then(response=>response.data);
    return {
        type:'ACCEPT_IN_REQ',
        payload:request
    }
}


// Delete user from friends //
export function deleteFriend(curUser,friend){
    const request = axios.post(`/api/deleteFriend`,{curUser,friend})
    .then(response=>response.data);
    return {
        type:'DELETE_FRIEND',
        payload:request
    }
}


// Get all friends of current user //
export function showFriends(userId){
    const request = axios.get(`/api/showFriends?userId=${userId}`)
    .then(response=>response.data);
    return {
        type:'GET_ALL_FRIENDS',
        payload:request
    }
}


// Clear registration form //
export function clearReg(){
    return{
        type:'CLEAR_REG',
        payload:{
            success:false,
            user:null
        }
    }
}


/*------------------------------------------ CHAT ACTIONS ------------------------------------------*/


// Load last 35 messages in room //
export function loadInitialData(room,user){
    return (dispatch) => {
        axios.get(`/api/loadInitialData?room=${room}&user=${user}`)
        .then(({data})=>{
            let response = {
                success:true,
                data
            }

            dispatch({
                type:'LOAD_MSGS',
                payload:response
            })
        }) 
    }
}


// Add new message to the current room //
export const addItem = (message) => ({
    type: "ADD_ITEM",
    payload: message
})


// Mark message as read //
export function readMessage(message){
    return (dispatch) => {
        axios.post(`/api/readMessage`,{message})
        .then(({data})=>{
            let response = {
                data
            }
            dispatch({
                type:'READ_MSG',
                payload:response
            })
        })  
    }
}


/*------------------------------------------ GOOGLE ACTIONS ------------------------------------------*/


// Register and sign-in //
export function googleRegLog(token) {
    const request = axios.post(`/api/googleRegLog`,{token})
    .then(response=>response.data);

    return {
        type:'GOOGLE_REG_LOG',
        payload:request
    }
}