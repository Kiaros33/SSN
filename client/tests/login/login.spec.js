import React from 'react';
import Login from '../../src/containers/login/login';
import * as actions from '../../src/actions/index';
import {Link,Route,MemoryRouter} from 'react-router-dom';
import {shallow,mount,render} from 'enzyme';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer'

describe('<Login/> tests',()=>{
    const initialState = {
        user:{
            login:{
                error:true,
                message:"You are not logged in"
            }
        }
    };
    const mockStore = configureStore([thunk]);
    let store = mockStore(initialState);

    it('should render correctly',() => {
        const wrapper = shallow(<Login store={store}/>);

        expect(wrapper.prop('store').getState().user).toEqual(initialState.user); //check initialStore
        const email = wrapper.dive().find('input.email');
        const pass = wrapper.dive().find('input.pass');
        const submit = wrapper.dive().find('button').first();

        email.simulate('change',{target:{value:'user@mail.ru'}}); //do call input handler , but does not change the state...
        pass.simulate('change',{target:{value:'Password123'}}); //do call input handler , but does not change the state...

        wrapper.setState({
            email:'user@mail.ru',
            password:'Password123'
        }); //changing state manually

        expect(wrapper.state().email).toBe('user@mail.ru'); //check the state
        expect(wrapper.state().password).toBe('Password123'); //check the state
    })

    it('should match the snapshot',() => {
        const renderedValue = renderer.create(
        <MemoryRouter>
            <Route path="/login">
                <Login store={store}/>
            </Route>
        </MemoryRouter>).toJSON();
        expect(renderedValue).toMatchSnapshot();
    })

    
})
