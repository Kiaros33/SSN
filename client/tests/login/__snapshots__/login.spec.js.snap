// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login/> tests should match the snapshot 1`] = `
<div
  className="l_container"
>
  <form
    onSubmit={[Function]}
  >
    <h2>
      Log-in
    </h2>
    <div
      className="form_element"
    >
      <input
        className="email"
        onChange={[Function]}
        placeholder="Enter an e-mail"
        type="email"
        value=""
      />
    </div>
    <div
      className="form_element"
    >
      <input
        className="pass"
        onChange={[Function]}
        placeholder="Enter a password"
        type="password"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Go in
    </button>
    <a
      href="/register"
      onClick={[Function]}
    >
      <button>
        Register now!
      </button>
    </a>
  </form>
  <div>
    <button
      disabled={true}
      onClick={[Function]}
      style={
        Object {
          "background": "rgb(220, 78, 65)",
          "border": "none",
          "borderRadius": "3px",
          "color": "#dddddd",
          "display": "block",
          "fontSize": "20px",
          "fontWeight": "300",
          "margin": "10px auto",
          "opacity": 0.6,
          "padding": "10px 52px",
          "width": "260px",
        }
      }
      type="button"
    >
      <span
        aria-hidden={true}
        className="fa fa-google"
      />
    </button>
  </div>
  <div
    className="l_error"
  >
    
  </div>
</div>
`;
